object_store/src/aws/mod.rs

```diff
@@ -1094,12 +1103,30 @@ impl AmazonS3Builder {
     }
 }
 
+#[cfg(feature = "aws_profile")]
+fn profile_region(profile: String) -> Option<String> {
+    use crate::aws::credential::RegionProvider;
+    use futures::executor::block_on;
+
+    let provider = profile::ProfileProvider::new(profile, None);
+
+    block_on(provider.get_region())
```

Comment:
```suggestion
    Handle::current().block_on(provider.get_region())
```

Using [`tokio::Handle`](https://docs.rs/tokio/latest/tokio/runtime/struct.Handle.html#method.block_on) might be safer, as I believe the upstream assumes tokio